### ISSUE TITLE
batch normalization

### ISSUE THREAD

**Issue by ih4cku** [ih4cku/blog#27]
Simple experiments of Batch Normalization on MNIST. The network set up follows the paper[1] which contains just 3 fully connected hidden layers.

Compared nonlinear units with Sigmoid and ReLU.

References:

[1] Ioffe, Sergey, and Christian Szegedy. "Batch normalization: Accelerating deep network training by reducing internal covariate shift." arXiv preprint arXiv:1502.03167 (2015).

**Comment by ih4cku** [comment]
# Sigmoid

![naive_nobn](https://cloud.githubusercontent.com/assets/3128875/15348377/c44fe0aa-1cfc-11e6-84f6-8b0cd8958b33.png)

![naive_bn](https://cloud.githubusercontent.com/assets/3128875/15348380/cb663e98-1cfc-11e6-9a99-14b3e9c4c41e.png)

**Comment by ih4cku** [comment]
# ReLU

![relu_nobn](https://cloud.githubusercontent.com/assets/3128875/15348468/ba92fb8c-1cfd-11e6-8112-72067924b8f7.png)

![relu_bn](https://cloud.githubusercontent.com/assets/3128875/15348471/bf15683e-1cfd-11e6-91b3-d3284bef84a2.png)

**Comment by ih4cku** [comment]
posts
- http://blog.csdn.net/malefactor/article/details/51476961
- http://blog.csdn.net/malefactor/article/details/51549771
- https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html
- [What does the gradient flowing through batch normalization looks like ?](http://cthorey.github.io./backpropagation/)

**Comment by ih4cku** [comment]
- The learned affine transform applied to these normalized activations allows the BN transform to represent the identity transformation and preserves the network capacity.
